### PR TITLE
Fix check to disable LTO by default

### PR DESCRIPTION
**Pull request by bradparker**
This got sassc-ruby working for me using Nix on Darwin.

### CODE REVIEW

**Comment by lrworth** [comment]
https://github.com/sass/sassc-ruby/issues/148 is the original issue for this

File: ext/extconf.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@
 # darwin nix clang doesn't support lto
 # disable -lto flag for darwin + nix
 # see: https://github.com/sass/sassc-ruby/issues/148
-enable_lto_by_default = (Gem::Platform.local.os == "darwin" && !ENV['NIX_CC'].nil?)
+enable_lto_by_default = (Gem::Platform.local.os == "darwin" && ENV['NIX_CC'].nil?)
```
Comment (lrworth) [inline ext/extconf.rb:30]: On closer inspection this check is a little weird. Firstly I'm pretty sure `nil?` is a Rails addition. Secondly this should enable LTO by default _unless_ we are using nix clang on darwin. Maybe:

```
enable_lto_by_default = !(Gem::Platform.local.os == "darwin" && ENV['NIX_CC'])
```

Comment (bradparker) [reply ext/extconf.rb:30]: Thanks for having a look at this :).

I _think_ `.nil?` "comes standard".

```
~ $ irb

WARNING: This version of ruby is included in macOS for compatibility with legacy software.
In future versions of macOS the ruby runtime will not be available by
default, and may require you to install an additional package.

irb(main):001:0> nil.nil?
=> true
irb(main):002:0> "".nil?
=> false
irb(main):003:0>
```

I agree the check is weird and I think your version makes more sense. It reads well: "Enable LTO by default when we're not on Darwin _and_ the nix wrapped C compiler will be used".

Comment (lrworth) [reply ext/extconf.rb:30]: Oh sorry about that, you're absolutely right!

Also just to make it more explicit, it looks like in `master` the only time LTO is automatically enabled is on Darwin with Nix clang, which is the opposite to what we want.

Comment (bradparker) [reply ext/extconf.rb:30]: Yep, sorry. The title of my branch / PR is weird :P. In my head it was "Fix check to disable the enabling of LTO by default" ... ergh. 

I wonder if there isn't a better way to do this in general. Like to detect if LTO will work without having to _know_ that it won't work for the C compiler wrapped by Nix on Darwin. Like, could it also not work with other C compilers? Is there some way to ask: "is LTO available?". 

Until then it'd be good to get something like this merged as I'm sure other Darwin + Nix using Ruby devs are having a head-scratching time.

**Comment by bolandrm** [comment]
@glebm could you please review this if you get a moment?

**Comment by pelletencate** [comment]
@bolandrm Care to release this as v2.4.1?

**Comment by ywen** [comment]
Could you please release this change? Thanks!

**Comment by ywen** [comment]
I double checked. Could you please release the changes in this PR? Thank you!

**Comment by lloeki** [comment]
That last comment was a year ago and the release did not happen 😢

FTR a workaround:

```
bundle config [--local] build.sassc --disable-lto
bundle install
```

**Comment by kalbasit** [comment]
@bolandrm I ran into this issue as well. Is there any way you can release a new version with the fix?

**Pull request merged**